test/PackageVersionsTest/E2EInstallerTest.php

```diff
@@ -184,6 +186,60 @@ public function testRemovingPluginWithNoDevDoesNotAttemptToGenerateVersions() :
         );
     }
 
+    public function testOnReadonlyFilesystemDoesNotGenerateClasses() : void
+    {
+        $this->createPackageVersionsArtifact();
+        $this->createArtifact();
+
+        $this->writeComposerJsonFile(
+            [
+                'name'         => 'package-versions/e2e-local',
+                'require-dev'  => ['ocramius/package-versions' => '1.0.0'],
+                'repositories' => [
+                    ['packagist' => false],
+                    [
+                        'type' => 'artifact',
+                        'url' => $this->tempArtifact,
+                    ],
+                ],
+            ],
+            $this->tempLocalComposerHome
+        );
+
+        $this->execComposerInDir('install', $this->tempLocalComposerHome);
+
+        unlink(
+            $this->tempLocalComposerHome . '/vendor/ocramius/package-versions/src/PackageVersions/Versions.php'
+        );
+
+        chmod(
```

Comment:
Would a `chmod`on the above file lead to the same effects?

Reply:
No, because it is ok if `Versions.php` file is not writable.
In that case, file will be replaced.